--- setup.py.orig	2017-02-02 18:47:26.996737000 +0800
+++ setup.py	2017-02-02 18:48:30.059687601 +0800
@@ -59,13 +59,15 @@
                 include_dirs_numpy)
 define_macros = []
 
+include_dirs += ['../lapack-3.6.1/LAPACKE/include']
+
 ## Modify extra_link_args_lapacke depending on systems
 if os.path.isfile("libopenblas.py"):
     # This is for travis-CI.
     from libopenblas import extra_link_args_lapacke
 else:
     # This is when lapacke is installed on system
-    extra_link_args_lapacke = ['-llapacke', '-llapack', '-lblas']
+    extra_link_args_lapacke = ['../lapack-3.6.1/liblapacke.a', '../lapack-3.6.1/liblapack.a', '../lapack-3.6.1/librefblas.a', os.environ['EBROOTIFORT']+'/lib/intel64/libifcore_pic.a']
 
 # For MacPort
 # % sudo port install gcc6
