--- Src/getmachine.orig	2019-06-13 14:21:09.264213000 +0800
+++ Src/getmachine	2019-06-13 14:45:10.228126000 +0800
@@ -8,7 +8,14 @@
 #  7/97 IRIX64 directory name modified to handle IRIX7
 #  2/99 Cray options added (IW)
 #
-#  Written by J-R. Hill, modified by JDG
+#  DEFS:
+#
+#  -DMPI    => triggers MPI compilation
+#  -DNOFOX  => do not compile FoX related subroutines
+#  -DPLUMED => compile with calls to PLUMED added
+#  -DCTRLC  => compile with calls to csignal to trap Control C escape sequence
+#
+#  Initially written by J-R. Hill, modified by JDG
 #
 
 makedir()
@@ -29,7 +36,7 @@
 os=`uname -s`
 case $os in
 #
-# Mac OS-X 
+# Mac OS-X using Absoft compiler
 #
   Darwin)  dir="${os}${debug}"
 	   makedir
@@ -42,19 +49,20 @@
 #	       echo 'OPT=-O ' > makefile
 #	       echo 'OPT1=' >> makefile
 #  gfortran
-#	       echo 'OPT=-O2 ' > makefile
-#	       echo 'OPT1=-O1' >> makefile
+	       echo 'OPT=-O1 ' > makefile
+	       echo 'OPT1=-O1' >> makefile
 #  g95
-	       echo 'OPT=-O ' > makefile
-	       echo 'OPT1=' >> makefile
-	       echo 'OPT2=-ffloat-store' >> makefile
+#	       echo 'OPT=-O ' > makefile
+#	       echo 'OPT1=' >> makefile
+#	       echo 'OPT2=-ffloat-store' >> makefile
+#	       echo 'BAGGER=' >> makefile
 	   else	
 	       echo 'OPT=' > makefile
 	       echo 'OPT2=-ffloat-store' >> makefile
-#  gfortran
-#               echo 'BAGGER=-g -Wall -Wunused-parameter -fbounds-check ' >> makefile
 #  g95
-               echo 'BAGGER=-g -Wall -Wunused-vars -Wuninitialized -fbounds-check -ftrace=full' >> makefile
+#               echo 'BAGGER=-g -Wall -Wuninitialized -fbounds-check -ftrace=full' >> makefile
+#  gfortran
+               echo 'BAGGER=-g -Wall -Wunused-parameter -fbounds-check ' >> makefile
 	   fi
            target=${1}_
 #  IBM
@@ -63,26 +71,35 @@
 #  Absoft
 #           echo 'RUNF90=f90 ' >> makefile
 #           echo 'FFLAGS=-I.. -B18 ' >> makefile
-#  gfortran
-#           echo 'RUNF90=gfortran ' >> makefile
 #  g95
-           echo 'RUNF90=g95 ' >> makefile
+#           echo 'RUNF90=mpif90 ' >> makefile
+#           echo 'RUNF90=g95 ' >> makefile
+#  gfortran
+           echo 'RUNF90=gfortran ' >> makefile
            echo 'FFLAGS=-I.. ' >> makefile
+#  C compiler
+           echo 'RUNCC=gfortran ' >> makefile
+           echo 'CFLAGS=-funroll-all-loops -ffast-math -O3' >> makefile
 #  IBM
 #           echo 'LIBS=' >> makefile
 #  Absoft
 #           echo 'LIBS=-lU77' >> makefile
-#  gfortran
-#           echo 'LIBS=' >> makefile
 #  g95
            echo 'LIBS=' >> makefile
+#           echo 'LIBS=-lmpi' >> makefile
 	   echo 'LDFLAGS=' >> makefile
            echo 'BLAS=blas.o' >> makefile
            echo 'LAPACK=lapack.o' >> makefile
            echo 'ETIME=' >> makefile
            echo 'GULPENV=' >> makefile
 	   echo 'CDABS=cdabs.o' >> makefile
-           echo 'DEFS=' >> makefile
+# If PLUMED is needed then set -DPLUMED
+#           echo 'DEFS=-DFLUSH -DPLUMED' >> makefile
+           echo 'DEFS=-DFLUSH ' >> makefile
+           echo 'PLUMEDDEFS=-DDL_POLY' >> makefile
+#           echo 'DEFS=-DFLUSH -DMPI' >> makefile
+           echo 'ARCHIVE = ar rcv' >> makefile
+           echo 'RANLIB = ar ts' >> makefile
            ;;
 #
 # This is an IBM RS/6000.
@@ -186,40 +203,45 @@
 	   then
 # PG compiler
 #	       echo 'OPT=-fast' > makefile
-#	       echo 'OPT1=-O1 ' >> makefile
-# Intel compiler
-#	       echo 'OPT=-O3 -mp1 ' > makefile
 #	       echo 'OPT1=-O1' >> makefile
-#	       echo 'OPT2=-O3 -mp1 ' >> makefile
+# Intel compiler
+	       echo 'OPT=-O3 -mp1 ' > makefile
+	       echo 'OPT1=-O1' >> makefile
+	       echo 'OPT2=-O3 -mp1 ' >> makefile
 # Absoft compiler
 #	       echo 'OPT=-O' > makefile
 #	       echo 'OPT1=-O' >> makefile
 #  g95
-	       echo 'OPT=-O ' > makefile
-	       echo 'OPT1=' >> makefile
-	       echo 'OPT2=-ffloat-store' >> makefile
+#	       echo 'OPT=-O ' > makefile
+#	       echo 'OPT1=' >> makefile
+#	       echo 'OPT2=-ffloat-store' >> makefile
 	       echo 'BAGGER=' >> makefile
 	   else	
 	       echo 'OPT=' > makefile
 #  g95
-	       echo 'OPT2=-ffloat-store' >> makefile
-               echo 'BAGGER=-g -Wall' >> makefile
+#	       echo 'OPT2=-ffloat-store' >> makefile
+#               echo 'BAGGER=-g -Wall' >> makefile
+#               echo 'BAGGER=-g ' >> makefile
+	       echo 'BAGGER=' >> makefile
 	   fi
            target=${1}_
 # PG compiler
 #  	   echo 'RUNF90=pgf90 -Bstatic' >> makefile
 # Intel compiler
-#  	   echo 'RUNF90=ifort ' >> makefile
+  	   echo 'RUNF90=ifort ' >> makefile
+           echo 'RUNCC=icc ' >> makefile
 # Absoft compiler
 #           echo 'RUNF90=f90' >> makefile
 # g95
-           echo 'RUNF90=g95 ' >> makefile
+#           echo 'RUNF90=g95 ' >> makefile
+#           echo 'RUNCC=gcc ' >> makefile
 # PG compiler
 #	   echo 'FFLAGS=-I.. ' >> makefile
 #	   echo 'LIBS=' >> makefile
 # Intel compiler
-#	   echo 'FFLAGS=-I.. ' >> makefile
-#	   echo 'LIBS=' >> makefile
+	   echo 'FFLAGS=-I.. ' >> makefile
+	   echo 'LIBS=' >> makefile
+	   echo 'CFLAGS=-I.. ' >> makefile
 # Absoft compiler
 #           echo 'FFLAGS=-I.. -B18 ' >> makefile
 #           echo 'LIBS=-lblas -lU77' >> makefile
@@ -231,14 +253,23 @@
 #	   echo 'LDFLAGS=-Bstatic' >> makefile
 # Intel compiler
 #	   echo 'LDFLAGS=-static' >> makefile
-#  g95
+# g95
 	   echo 'LDFLAGS=' >> makefile
+#
+# General options
+#
            echo 'ETIME=' >> makefile
            echo 'BLAS=blas.o' >> makefile
+# for chemshell use, we should have lapack so comment the following line out
            echo 'LAPACK=lapack.o' >> makefile
            echo 'GULPENV=' >> makefile
 	   echo 'CDABS=cdabs.o' >> makefile
-           echo 'DEFS=' >> makefile
+# If PLUMED is needed then set -DPLUMED
+#           echo 'DEFS=-DFLUSH -DPLUMED' >> makefile
+           echo 'DEFS=-DFLUSH ' >> makefile
+           echo 'PLUMEDDEFS=-DDL_POLY' >> makefile
+           echo 'ARCHIVE = ar rcv' >> makefile
+           echo 'RANLIB = ar ts' >> makefile
            ;;
 #
 # This is OSF1
@@ -371,7 +402,17 @@
 #then
 #  strip gulp
 #fi
+
+if [ "$target" = "lib_" ]
+then
+#
+# Move library to top of GULP tree
+#
+mv libgulp.a ../../libgulp.a
+elif [ "$target" = "gulp_" ]
+then
 #
 # Move executable back to level at which make command was issued
 #
 mv gulp ../gulp
+fi
