diff -rw -u qe-6.4.1.orig/Modules/set_hubbard_l.f90 qe-6.4.1/Modules/set_hubbard_l.f90
--- qe-6.4.1.orig/Modules/set_hubbard_l.f90	2019-04-08 20:09:54.000000000 +0800
+++ qe-6.4.1/Modules/set_hubbard_l.f90	2020-03-01 13:47:27.513986884 +0800
@@ -23,7 +23,7 @@
      !
      CASE( 'Ti', 'V',  'Cr', 'Mn', 'Fe', 'Co', 'Ni', 'Cu', 'Zn', &
            'Zr', 'Nb', 'Mo', 'Tc', 'Ru', 'Rh', 'Pd', 'Ag', 'Cd', &
-           'Hf', 'Ta', 'W',  'Re', 'Os', 'Ir', 'Pt', 'Au', 'Hg'  )
+           'Hf', 'Ta', 'W',  'Re', 'Os', 'Ir', 'Pt', 'Au', 'Hg', 'La'  )
         !
         hubbard_l = 2  
         !
diff -rw -u qe-6.4.1.orig/Modules/set_hubbard_n.f90 qe-6.4.1/Modules/set_hubbard_n.f90
--- qe-6.4.1.orig/Modules/set_hubbard_n.f90	2019-04-08 20:09:54.000000000 +0800
+++ qe-6.4.1/Modules/set_hubbard_n.f90	2020-03-01 17:06:39.201985242 +0800
@@ -42,7 +42,7 @@
      !  
         hubbard_n = 4 
      !  ... rare earths (actinoids )
-     CASE ('Th','Pa','U', 'Np','Pu','Am','Cm','Bk','Cf','Es','Fm','Md','No','Lr' )
+     CASE ('La','Th','Pa','U', 'Np','Pu','Am','Cm','Bk','Cf','Es','Fm','Md','No','Lr' )
         !
         hubbard_n = 5
         !
diff -rw -u qe-6.4.1.orig/PW/src/tabd.f90 qe-6.4.1/PW/src/tabd.f90
--- qe-6.4.1.orig/PW/src/tabd.f90	2019-04-08 20:09:54.000000000 +0800
+++ qe-6.4.1/PW/src/tabd.f90	2020-03-02 00:31:22.920069822 +0800
@@ -53,6 +53,8 @@
      !
      ! RARE EARTHS
      !
+     CASE( 'La' )
+        hubbard_occ = 1.d0
      CASE( 'Ce', 'Th' )
         hubbard_occ = 2.d0
      !
