--- CMakeLists.txt.orig	2020-08-06 14:01:14.294671088 +0800
+++ CMakeLists.txt	2020-08-06 16:50:28.027057542 +0800
@@ -37,21 +37,26 @@
     if (NOT TRIQS_PATH)
         message(FATAL_ERROR "Set TRIQS_PATH to the install directory of TRIQS (e.g. /opt/triqs).")
     endif()
+    if (NOT TRIQS_DFTTOOLS_PATH)
+        message(FATAL_ERROR "Set TRIQS_DFTTOOLS_PATH to the install directory of TRIQS DFTTools (e.g. /opt/triqs).")
+    endif()
     set(TRIQS_SITE_PACKAGES ${TRIQS_PATH}/lib/python2.7/site-packages)
+    set(TRIQS_DFTTOOLS_SITE_PACKAGES ${TRIQS_DFTTOOLS_PATH}/lib/python2.7/site-packages)
     set(ENV{PYTHONPATH} ${TRIQS_SITE_PACKAGES}:$ENV{PYTHONPATH})
+    set(ENV{PYTHONPATH} ${TRIQS_DFTTOOLS_SITE_PACKAGES}:$ENV{PYTHONPATH})
 
     # Check version of TRIQS
     execute_process(COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/cmake/show_triqs_version.py OUTPUT_VARIABLE TRIQS_VERSION)
     if (TRIQS_VERSION STREQUAL "NOT_FOUND")
         message(FATAL_ERROR "TRIQS not found! Set TRIQS_PATH properly.")
-    elseif (TRIQS_VERSION STREQUAL "1.4" OR TRIQS_VERSION MATCHES "^2.1.")
+    elseif (TRIQS_VERSION STREQUAL "1.4" OR TRIQS_VERSION MATCHES "^2.1")
         message(STATUS "Found TRIQS Version ${TRIQS_VERSION}")
     else()
         message(WARNING "TRIQS Version ${TRIQS_VERSION} may be imcompatible with DCore.")
     endif()
 
     # Check if DFTools is installed.
-    get_filename_component(DFTToolsPath1 "${TRIQS_SITE_PACKAGES}/triqs_dft_tools" REALPATH "${CMAKE_BINARY_DIR}")
+    get_filename_component(DFTToolsPath1 "${TRIQS_DFTTOOLS_SITE_PACKAGES}/triqs_dft_tools" REALPATH "${CMAKE_BINARY_DIR}")
     get_filename_component(DFTToolsPath2 "${TRIQS_SITE_PACKAGES}/pytriqs/applications/dft" REALPATH "${CMAKE_BINARY_DIR}")
     if ( (NOT IS_DIRECTORY ${DFTToolsPath1}) AND (NOT IS_DIRECTORY ${DFTToolsPath2}))
         message(FATAL_ERROR "TRIQS/DFTTools not found! Please install TRIQS/DFTTools first.")
